src/pages/SignUp/SignUp.js

```diff
@@ -1,10 +1,354 @@
 import React, { Component } from 'react';
+import './SignUp.scss';
+import { AiFillLock } from 'react-icons/ai';
+import { FaUserLock } from 'react-icons/fa';
+import {AiOutlineUnlock} from 'react-icons/ai';
+import { CHECK_VALIDATION_API } from '../../config';
+import { DUPLICATION_CHECK_API } from '../../config';
+import { POST_EMAIL_API } from '../../config';
+import { CODE_CHECK_API } from '../../config';
+
 
 class SignUp extends Component {
+    constructor() {
+        super();
+        this.state = {
+            id: '',
+            pw: '',
+            name: '',
+            phone:'',
+            email: '',
+            passwordcheck: '',
+            idDoubleCheck: false,
+            codeNumber: '',
+
+            validID: true,
+            validPW: true,
+            validDoubleCheck: true,
+            validName: true,
+            validPhone: true,
+            validEmail: true,
+           
+
+            allNecessaryChecked: false,
+            checked1: false,
+            checked2: false,
+            checked3: false,
+
+            validCode: false,
+        }
+    }
+
+    validationCheck = (name, value) => {
+        if (name ==='id') {
+            let idReg = /^[a-z]+[a-z0-9]{5,19}$/g;
+            const validID = idReg.test(value);
+            this.setState({validID});
+            return;
+        } 
+        if (name ==='pw') {
+            let passwordRules = /^(?=.*[a-zA-Z])(?=.*[!@#$%^*+=-])(?=.*[0-9]).{8,16}$/;
+            const validPW = passwordRules.test(value);
+            this.setState({validPW});
+            return;   
+        } 
+        if (name === 'passwordcheck') {
+            const validDoubleCheck = this.state.pw === value
+            this.setState({validDoubleCheck});
+            return;
+        } 
+        if (name === 'name') {
+            let reg = /^[가-힣]{2,4}$/;
+            const validName = reg.test(value);
+            this.setState({validName});
+            return;
+        } 
+        if (name === 'phone') {
+            let regPhone = /^\d{3}-\d{3,4}-\d{4}$/;
+            const validPhone = regPhone.test(value);
+            this.setState({validPhone});
+            return;
+        } 
+        if (name === 'email') {
+            let emailRule = /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i;
+            const validEmail = emailRule.test(value);
+            this.setState({validEmail});
+            return;
+        } 
+    }
+
+    handleInputValueChange = (e) => {
+        const {name, value} = e.target;
+        this.validationCheck(name, value);
+        this.setState({[name]: value});
+    }
+
+    checkValidation = (e) => {
+        e.preventDefault();
+        const {id, pw, name, phone, email, passwordcheck, idDoubleCheck, validCode} = this.state;
+        if(!idDoubleCheck){
+            alert("아이디 중복확인을 해주세요")       
+            return;
+        }
+        if(!validCode){
+            alert("이메일 인증을 진행해주세요")
+            return;
```

Comment:
```suggestion
        if(!idDoubleCheck){
            return alert("아이디 중복확인을 해주세요")       
        }
        if(!validCode){
            return alert("이메일 인증을 진행해주세요")
```
이렇게 리팩토링 가능할 것 같습니다!